src/test/ui/macros/assert-trailing-junk.rs

```diff
@@ -0,0 +1,14 @@
+// Ensure assert macro does not ignore trailing garbage.
+//
+// See https://github.com/rust-lang/rust/issues/60024 for details.
+
+fn main() {
+    assert!(true some extra junk, "whatever");
+    //~^ ERROR expected one of
+
+    assert!(true some extra junk);
+    //~^ ERROR expected one of
+
+    assert!(true, "whatever" blah);
+    //~^ ERROR no rules expected
```

Comment:
Is `assert!(foo, bar, baz)` supposed to be accepted? I think you'll need to also check how many expressions have been found so far and error if more than 2.

Reply:
I thought about this when implementing, but it is accepted. Second to last arguments represent format string + its arguments.

```rust
assert!(foo == 3, "foo is expected to be 3 but is {}", foo);
```
produces
```
thread 'main' panicked at 'foo is expected to be 3 but is 2', test.rs:3:5
```

Reply:
Ah! I see, you're right. Disregard me.